adam-core/src/main/scala/org/bdgenomics/adam/rdd/features/FeatureRDDFunctions.scala

```diff
@@ -133,11 +132,167 @@ class FeatureRDDFunctions(featureRDD: RDD[Feature]) extends Serializable with Lo
+      val seqname = feature.getContigName
+      val source = Option(feature.getSource).getOrElse(".")
+      val featureType = Option(feature.getFeatureType).getOrElse(".")
+      val start = feature.getStart + 1 // GTF/GFF ranges are 1-based
+      val end = feature.getEnd // GTF/GFF ranges are closed
+      val score = Option(feature.getScore).getOrElse(".")
+      val strand = Features.toString(feature.getStrand)
+      val frame = Option(feature.getFrame).getOrElse(".")
+      val attributes = Features.gatherAttributes(feature).map(escape).mkString("; ")
+      List(seqname, source, featureType, start, end, score, strand, frame, attributes).mkString("\t")
+    }
+    featureRDD.map(toGtf).saveAsTextFile(fileName)
+  }
+
+  def saveAsGff3(fileName: String) = {
+    def escape(entry: (Any, Any)): String = {
+      entry._1 + "=" + entry._2
+    }
+
+    def toGff3(feature: Feature): String = {
+      val seqid = feature.getContigName
+      val source = Option(feature.getSource).getOrElse(".")
+      val featureType = Option(feature.getFeatureType).getOrElse(".")
+      val start = feature.getStart + 1 // GFF3 coordinate system is 1-based
+      val end = feature.getEnd // GFF3 ranges are closed
+      val score = Option(feature.getScore).getOrElse(".")
+      val strand = Features.toString(feature.getStrand)
+      val phase = Option(feature.getPhase).getOrElse(".")
+      val attributes = Features.gatherAttributes(feature).map(escape).mkString(";")
+      List(seqid, source, featureType, start, end, score, strand, phase, attributes).mkString("\t")
+    }
+    featureRDD.map(toGff3).saveAsTextFile(fileName)
+  }
+
+  def saveAsBed(fileName: String) = {
+    def toBed(feature: Feature): String = {
+      val chrom = feature.getContigName
+      val start = feature.getStart
+      val end = feature.getEnd
+      val name = Features.nameOf(feature)
+      val score = Option(feature.getScore).getOrElse(".")
+      val strand = Features.toString(feature.getStrand)
+
+      if (!feature.getAttributes.containsKey("thickStart")) {
+        // write BED6 format
+        List(chrom, start, end, name, score, strand).mkString("\t")
+      } else {
+        // write BED12 format
+        val thickStart = feature.getAttributes.getOrElse("thickStart", ".")
+        val thickEnd = feature.getAttributes.getOrElse("thickEnd", ".")
+        val itemRgb = feature.getAttributes.getOrElse("itemRgb", ".")
+        val blockCount = feature.getAttributes.getOrElse("blockCount", ".")
+        val blockSizes = feature.getAttributes.getOrElse("blockSizes", ".")
+        val blockStarts = feature.getAttributes.getOrElse("blockStarts", ".")
+        List(chrom, start, end, name, score, strand, thickStart, thickEnd, itemRgb, blockCount, blockSizes, blockStarts).mkString("\t")
+      }
+    }
+    featureRDD.map(toBed).saveAsTextFile(fileName)
+  }
+
+  def saveAsIntervalList(fileName: String) = {
+    def escape(entry: (Any, Any)): String = {
```

Comment:
Where does `escape` get used? Perhaps I'm just missing things but I don't see it get called anywhere.